elastic/kibana
# [Security Solution][Detections] Fixes Rule Details when viewing deleted Rule from Alert with flattened structure

## Summary

Resolves https://github.com/elastic/kibana/issues/122013 by transforming flattened `alertHit` back to a `Rule` using the existing `expandDottedObject` utility (which was moved to `security_solution/common/utils` for global use). Also fixed margins on deleted badge (and ensured no overlap when title overflows).

<p align="center">
  <img width="500" src="https://user-images.githubusercontent.com/2946766/148868172-f8af8340-f49c-46e3-8cd6-4ebb38d7d76f.png" />
</p>



## Test Instructions:
* Create Rule with all available fields filled out
* Generate alerts for Rule
* Delete Rule and navigate to Rule Details from Alert Details
* Verify `Deleted` badge is present and Rule Details are filled out again

Test with 7.x and 8.x alerts (as backwards compatibility was kept)
